src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -80,6 +82,29 @@ private static List<Group> FindGroups(string path, string query, string template
         match = Regex.Match($"{testedPath}/", regexPattern);
         return match.Groups.Cast<Group>().Skip(1).ToList();
     }
+    
+    /// <summary>
+    /// The placeholders that are not placed at the end of the template are delimited by forward slashes, only the last one, the catch-all can match more segments.
+    /// </summary>
+    /// <param name="escapedTemplate">The escaped path template.</param>
+    /// <returns>The pattern for values replacement.</returns>
+    private static string GenerateRegexPattern(string escapedTemplate)
+    {
+        // First we count the matches
+        var placeHoldersCountMatch = RegexPlaceholders().Matches(escapedTemplate);
+        int index = 0, placeHoldersCount = placeHoldersCountMatch.Count;
+
+        // We know that the replace process will be started from the beginning of the url,
+        // so we can use a simple counter to determine the last placeholder
+        string MatchEvaluator(Match match)
+        {
+            var groupName = match.Groups[1].Value;
+            index++;
+            return index == placeHoldersCount ? $"(?<{groupName}>[^&]*)" : $"(?<{groupName}>[^/|&]*)";
+        }
+
+        return $@"^{RegexPlaceholders().Replace(escapedTemplate, MatchEvaluator)}";
```

Comment:
Is it possible to move this replacement to the static object? It appears not, due to the varying `escapedTemplate` that represents the current path template. This is unfortunate 😢 as it seems there are no options available.